src/NearestNeighborsModule.jl

```diff
@@ -0,0 +1,65 @@
+module NearestNeighborsModule
+
+using KernelFunctions, LinearAlgebra, SparseArrays, AbstractGPs
+
+function make_B(pts::AbstractVector{T}, k::Int, kern::Kernel) where {T}
```

Comment:
Could you please add a comment explaining what this function does.

Additionally, is there a more descriptive name possible than `make_B`? As someone who is aware that this approximation exists, but hasn't worked with it previously, I have no idea what `make_B` might mean haha

Reply:
An excellent point! I have added a little documentation, which hopefully should make things clearer.